engine/spec/components/targeted_content_spec.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples "public_targeted_content" do
+  it "renders" do
+    expect(component.at(".cads-targeted-content")).to be_present
+  end
+
+  it "has no label" do
+    expect(component.at(".cads-badge")).not_to be_present
+  end
+end
+
+RSpec.shared_examples "adviser_targeted_content" do
+  let(:badge) { component.at(".cads-badge") }
+
+  it "renders an adviser variant" do
+    expect(component.at(".cads-targeted-content--adviser")).to be_present
+  end
+
+  it "has adviser label" do
+    expect(badge.text).to eq "Adviser"
+  end
+
+  context "when welsh language" do
+    before { I18n.locale = :cy }
+    it "has translated label" do
+      expect(badge.text).to eq "Cynghorydd"
+    end
+  end
+end
+
+RSpec.describe CitizensAdviceComponents::TargetedContent, type: :component do
+  subject(:component) do
+    render_inline(
+      described_class.new(
+        id: "targeted-content-example",
+        type: type.presence,
+        title: title.presence,
+        heading_level: heading_level.presence
+      )
+    ) do
+      "Example content"
+    end
+  end
+
+  let(:type) { :public }
+  let(:title) { "Example title" }
+  let(:heading_level) { nil }
+
+  it "renders content block" do
+    expect(component.text).to include "Example content"
+  end
+
+  it_behaves_like "public_targeted_content"
+
+  it "has a h2 title by default" do
+    expect(component.at("h2").text.strip).to eq(title)
+  end
+
+  context "when missing type" do
+    let(:type) { nil }
+
+    context "production rails env" do
```

Comment:
should we add a context for non-production if the behaviour is diff?

Reply:
Yeah good shout. This is technically covered by the behaviour in `fetch_or_fallback` but doesn't hurt to be explicit.